src/main/java/gyro/google/compute/InstanceResource.java

```diff
@@ -0,0 +1,316 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Instance;
+import com.google.api.services.compute.model.InstancesSetLabelsRequest;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+
+/**
+ * Creates a network.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *      zone: "us-west1-a"
+ *
+ *      google::instance gyro-dev-1
+ *          name: "gyro-development"
+ *          description: "Testing for Gyro"
+ *          zone: $zone
+ *          machine-type: "zones/$(zone)/machineTypes/n1-standard-1"
+ *
+ *          network-interfaces
+ *              network: $(external-query google::compute-network {name: "default"})
+ *          end
+ *
+ *          initialize-disks
+ *              boot: true
+ *
+ *              initialize-params
+ *                  disk-name: "gyro-boot-disk"
+ *                  source-image: "projects/debian-cloud/global/images/family/debian-9"
+ *              end
+ *          end
+ *
+ *          initialize-disks
+ *              initialize-params
+ *                  disk-name: "gyro-secondary-disk"
+ *                  source-image: "projects/debian-cloud/global/images/family/debian-9"
+ *              end
+ *          end
+ *
+ *          labels: {
+ *              "gyro": "install"
+ *          }
+ *      end
+ */
+@Type("instance")
+public class InstanceResource extends ComputeResource implements Copyable<Instance> {
+
+    private String name;
+    private String zone;
+    private String description;
+    private String machineType;
+    private List<InstanceNetworkInterface> networkInterfaces;
+    private List<InstanceAttachedDisk> disks;
+    private String selfLink;
+    private Map<String, String> labels;
+    private String labelFingerprint;
+
+    private List<InstanceAttachedDisk> initializeDisks;
+
+    /**
+     * The name of the resource when initially creating the resource. Must be 1-63 characters, first character must be a lowercase letter and all following characters must be a dash, lowercase letter, or digit, except the last character, which cannot be a dash.
+     */
+    @Regex("[a-z]([-a-z0-9]*[a-z0-9])?")
```

Comment:
this regex does not include the 1-36 character limiter.

Reply:
Updated regex to `^[a-z]([-a-z0-9]{1,61}[a-z0-9])?`